core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -99,20 +100,36 @@ type CustomizationArgsNameAndValueArray = {
   // Finally, use [keyof ComponentSpecsType] to create a union.
 }[keyof ComponentSpecsType];
 
+// RteComponentId extracts the frontend_id array from
+// the component in ComponentSpecsType.
+export type RteComponentId = {
+  [K in keyof ComponentSpecsType]:
+  ComponentSpecsType[K]['frontend_id'];
+}[keyof ComponentSpecsType];
+
 @Component({
   selector: 'oppia-rte-helper-modal',
   templateUrl: './rte-helper-modal.component.html',
 })
 export class RteHelperModalComponent {
+  @Input() componentId: RteComponentId;
   @Input() customizationArgSpecs: CustomizationArgsSpecsType;
   @Input() attrsCustomizationArgsDict: CustomizationArgsForRteType;
   modalIsLoading: boolean = true;
-  currentRteIsMathExpressionEditor: boolean = false;
-  currentRteIsLinkEditor: boolean = false;
+  saveButtonIsDisabled: boolean = false;
+  errorMessage: string;
   tmpCustomizationArgs: CustomizationArgsNameAndValueArray = [];
   @ViewChild('schemaForm') schemaForm!: NgForm;
   defaultRTEComponent: boolean;
   public customizationArgsForm: FormGroup;
+  rteFieldValues: Subscription | undefined;
```

Comment:
If we are always doing the subscription then why are we allowing the possibility of undefined here?

Reply:
done